### DISCUSSION TITLE
Device blocked

### DISCUSSION THREAD

**Discussion by nadja75** [blackmagic-debug/blackmagic#1758] [Q&A]
Good evening, I wrote the latest version of the blackmagic_allinone-stlink.bin file, on an STLink V2 device equipped with a Geehy APM32F103CBT6 chip, through the STL Link Utility software with SWD connection, and everything worked perfectly, my device is now seen as Black Magic DFU, Black Magic Trace Capture and two USB-Serial devices. Now the problem I'm having is that I can no longer connect to the device in SWD mode with any software available, not even through Segger's J-Link. I believe the new firmware has changed the CLK and DIO pins set by default by the manufacturer. How can I restore the device? 

![Senza titolo](https://github.com/blackmagic-debug/blackmagic/assets/72427253/00cb03df-345f-4e6b-9d55-65efd8adb50e)

**Comment by ALTracer** [comment]
> blackmagic_allinone-stlink.bin

That filename sounds familiar yet unsupported. None of blackmagic-debug org CI workflows produce firmware binaries with such a filename. Are you perhaps seeking support from koendv's fork?

(`stlink` pinouts leave it usable, but) `swlink` pinouts (as found on SWIM STM8S-Discovery and STM32VLDiscovery stlink/v1, but also often used for random bluepills) unmap F103C8's own SWJ-DP from PA13/14 and repurpose it for GPIO bitbanging. If you're positive you've flashed `stlink`, then you may have faced some Apex Micro incompatibility, even though APM32F103 should behave similar enough for BMF purposes.

In any case, general unbricking directions apply -- use Connect Under Reset (the upstream debug adapter literally pulls down nRST pin of target and pokes the Debug bus, setting up vector catch on Reset handler etc. saving it from RCC runaway and SWJ-DP unmapping). Or pull up BOOT0 and reboot, activating both SWJ-DP and AN2606 System Memory (MaskROM) bootloader on USART1. No need to rely on JLink, it's not a panacea.

**Reply by nadja75** [reply]
Yes, you are right, the file comes from the koendv fork. I had tried to compile the latest version of the official fork but unfortunately I received an error when writing the ELF file (presumably due to lack of memory on the ST chip). Unfortunately for BOT0 I have no solution, my device is a white STLink V2, therefore not equipped with this switch. I will try your solutions and update you. For now I thank you.

![20240130_184046](https://github.com/blackmagic-debug/blackmagic/assets/72427253/717877d4-dffb-4428-97ff-1ea99cecf39e)

**Reply by nadja75** [reply]
Perfect, everything resolved, I deleted the R1 and the device went into BOOT MODE. Flashed again with appropriate firmware and reinstalled the R1.
Thank you for your suggestion.

**Reply by dragonmux** [reply]
In regards the .elf from CI, if you are using `stlink-tool` it does not understand the ELF file format, it assumes the file is a raw firmware binary and will write the entire file contents verbatim - for this tool you must use the `.bin` files. ELFs contain sections, headers, debug information, etc, which is why those don't fit. The ones built in CI are also built for the project's bootloader, not the ST bootloader. For ST-Link builds to go on with the default bootloader, you either have to `make PROBE_HOST=stlink ST_BOOTLOADER=1` or `meson setup build --cross-file=cross-file/stlink.ini -Dbmd_bootloader=false && meson compile build`.

With regards to interacting with the firmware: BMD provides a complete solution with no external tools required - the two serial ports are the GDB server, and the debug connector serial port. The J-Link tools talk their own proprietary protocol with J-Link adaptors and are entirely incompatible with this. To talk with BMD you need to [follow the getting started guide](https://black-magic.org/getting-started.html).
